arithmetization/src/main/java/net/consensys/linea/zktracer/module/oob/Oob.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package net.consensys.linea.zktracer.module.oob;
+
+import static net.consensys.linea.zktracer.types.Conversions.bigIntegerToBytes;
+
+import java.math.BigInteger;
+import java.nio.MappedByteBuffer;
+import java.util.List;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import net.consensys.linea.zktracer.ColumnHeader;
+import net.consensys.linea.zktracer.container.stacked.list.StackedList;
+import net.consensys.linea.zktracer.module.Module;
+import net.consensys.linea.zktracer.module.add.Add;
+import net.consensys.linea.zktracer.module.hub.Hub;
+import net.consensys.linea.zktracer.module.mod.Mod;
+import net.consensys.linea.zktracer.module.wcp.Wcp;
+import net.consensys.linea.zktracer.opcode.OpCode;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.internal.Words;
+
+@RequiredArgsConstructor
+/** Implementation of a {@link Module} for out of bounds. */
+public class Oob implements Module {
+
+  /** A list of the operations to trace */
+  @Getter private final StackedList<OobChunk> chunks = new StackedList<>();
+
+  private final Hub hub;
+  private final Add add;
+  private final Mod mod;
+  private final Wcp wcp;
+
+  @Override
+  public String moduleKey() {
+    return "OOB";
+  }
+
+  static final List<Address> PRECOMPILES_HANDLED_BY_OOB =
+      List.of(
+          Address.ECREC,
+          Address.SHA256,
+          Address.RIPEMD160,
+          Address.ID,
+          Address.ALTBN128_ADD,
+          Address.ALTBN128_MUL,
+          Address.ALTBN128_PAIRING,
+          Address.BLAKE2B_F_COMPRESSION);
+
+  @Override
+  public void tracePreOpcode(MessageFrame frame) { // This will be renamed to tracePreOp
+    this.chunks.add(new OobChunk(frame, add, mod, wcp, hub, false, 0, 0));
+    OpCode opCode = OpCode.of(frame.getCurrentOperation().getOpcode());
+
+    if (hub.pch().exceptions().none() && hub.pch().aborts().none() && opCode.isCall()) {
```

Comment:
the OOB is already trigger only when the right trigger condition is met imho. So only need `opCode.isCall()`

Reply:
Ok, I'll kill the first two checks.